src/test/java/org/zalando/nakadi/enrichment/MetadataEnrichmentStrategyTest.java

```diff
@@ -91,6 +91,34 @@ public void setFlowId() throws Exception {
     }
 
     @Test
+    public void whenFlowIdIsPresentDoNotOverride() throws Exception {
+        final EventType eventType = buildDefaultEventType();
+        final JSONObject event = buildBusinessEvent();
+        event.getJSONObject("metadata").put("flow_id", "something");
+        final BatchItem batch = createBatchItem(event);
+
+        final String flowId = randomString();
```

Comment:
I generated an id so to prove that it's not used. Makes sense? The idea is to show that even though there is a flow id for this thread, it's no longer used in case the value is already specified in the event.